source/PrintDiamond.h

```diff
@@ -0,0 +1,29 @@
+#ifndef PRINTDIAMOND_INCLUDED
+#define PRINTDIAMOND_INCLUDED
+
+#include<array>
+#include<vector>
+#include<iostream>
+
+int answer();
+
+class PrintDiamondClass
+{
+	public:
```

Comment:
You should generally make everything you can private, unless it **needs** to be publicly visible. In this example, all your functions must be public because they are called from outside. The arrays and ints however should be private because they are noone else's business.

Generally speaking, if you find yourself creating a member variable public, think hard about whether that is the right approach, as that means it can be modified by anyone at any time.
If you do have properties that must be changeable (or just readable) from the outside, convention is to have them private and add public functions to access them like so:
```c++
class foo
{
    public:
    int get_bar() { return m_bar; } // We generally don't put code in header files. I only did it here to demonstrate the point.
    void set_bar(int newBar) { m_bar = newBar; }

    private:
    int m_bar;
};
```